Add directions for testing self-signed certificates

zhiweicen opened the pull request:
Self-signed certificates have the benefit of not relying on a CA but can still provide certain security of the control plane network. This document describes the steps needed to test if a switch supports self-signed certificates.

shivarammysore commented:
This is good and very useful info.

Is storing certs under /tmp good?  /tmp is cleaned out at reboots.

Also, other things to consider:

   1. OVS stores certs under /var/lib/openvswitch and /etc/openvswitch
   directories.  Would storing these under those be good?
   2. When Controller certs are provided to Faucet, they must have the same
   permission of user who is running faucet service - normally, this could be
   "root" or "faucet".  If faucet, then chowm cert files to facuet:faucet (if
   faucet is user and faucet is group)
   3. One other thing to add is if we have more than 1 switch connecting
   via TLS to the same controller, how to add certs to the 2nd switch so that
   it can work with the already running controller.


Thanks

On Thu, Jul 13, 2017 at 6:04 PM, Josh Bailey <notifications@github.com>
wrote:

> Merged #756 <https://github.com/REANNZ/faucet/pull/756>.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/REANNZ/faucet/pull/756#event-1163570955>, or mute the
> thread
> <https://github.com/notifications/unsubscribe-auth/ABmf82YJ9A9lTWECwxeoR6Bf8LoafEL1ks5sNr6CgaJpZM4OXr4X>
> .
>

zhiweicen commented:
Thanks for the comments.

I've made another PR to make some improvements.

> Is storing certs under /tmp good? /tmp is cleaned out at reboots.
My original intention was to write the guidelines as instructions for
testing purpose. I realized somebody may want to run this for real.
Yes It's good to support those as well. I updated the paths in the PR.

> Also, other things to consider:
>
> 1. OVS stores certs under /var/lib/openvswitch and /etc/openvswitch
> directories. Would storing these under those be good?
Instead of using OVS folders I used /etc/ryu folder instead as this is
not strictly related to OVS.

> 2. When Controller certs are provided to Faucet, they must have the same
> permission of user who is running faucet service - normally, this could be
> "root" or "faucet". If faucet, then chowm cert files to facuet:faucet (if
> faucet is user and faucet is group)
Added a comment reminding the user doing that.

> 3. One other thing to add is if we have more than 1 switch connecting
> via TLS to the same controller, how to add certs to the 2nd switch so that
> it can work with the already running controller.
Added a section with the instructions to support that.

Zhiwei

On Thu, Jul 13, 2017 at 10:47 PM, shivarammysore
<notifications@github.com> wrote:
>
> This is good and very useful info.
>
> Is storing certs under /tmp good? /tmp is cleaned out at reboots.
>
> Also, other things to consider:
>
> 1. OVS stores certs under /var/lib/openvswitch and /etc/openvswitch
> directories. Would storing these under those be good?
> 2. When Controller certs are provided to Faucet, they must have the same
> permission of user who is running faucet service - normally, this could be
> "root" or "faucet". If faucet, then chowm cert files to facuet:faucet (if
> faucet is user and faucet is group)
> 3. One other thing to add is if we have more than 1 switch connecting
> via TLS to the same controller, how to add certs to the 2nd switch so that
> it can work with the already running controller.
>
>
> Thanks
>
> On Thu, Jul 13, 2017 at 6:04 PM, Josh Bailey <notifications@github.com>
> wrote:
>
> > Merged #756 <https://github.com/REANNZ/faucet/pull/756>.
> >
> > —
> > You are receiving this because you are subscribed to this thread.
> > Reply to this email directly, view it on GitHub
> > <https://github.com/REANNZ/faucet/pull/756#event-1163570955>, or mute the
> > thread
> > <https://github.com/notifications/unsubscribe-auth/ABmf82YJ9A9lTWECwxeoR6Bf8LoafEL1ks5sNr6CgaJpZM4OXr4X>
> > .
> >
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.

shivarammysore commented:
Thanks Zhiwei.


> On Jul 14, 2017, at 3:44 PM, zhiweicen <notifications@github.com> wrote:
> 
> Thanks for the comments.
> 
> I've made another PR to make some improvements.
> 
> > Is storing certs under /tmp good? /tmp is cleaned out at reboots.
> My original intention was to write the guidelines as instructions for
> testing purpose. I realized somebody may want to run this for real.
> Yes It's good to support those as well. I updated the paths in the PR.
> 
> > Also, other things to consider:
> >
> > 1. OVS stores certs under /var/lib/openvswitch and /etc/openvswitch
> > directories. Would storing these under those be good?
> Instead of using OVS folders I used /etc/ryu folder instead as this is
> not strictly related to OVS.
> 
> > 2. When Controller certs are provided to Faucet, they must have the same
> > permission of user who is running faucet service - normally, this could be
> > "root" or "faucet". If faucet, then chowm cert files to facuet:faucet (if
> > faucet is user and faucet is group)
> Added a comment reminding the user doing that.
> 
> > 3. One other thing to add is if we have more than 1 switch connecting
> > via TLS to the same controller, how to add certs to the 2nd switch so that
> > it can work with the already running controller.
> Added a section with the instructions to support that.
> 
> Zhiwei
> 
> On Thu, Jul 13, 2017 at 10:47 PM, shivarammysore
> <notifications@github.com> wrote:
> >
> > This is good and very useful info.
> >
> > Is storing certs under /tmp good? /tmp is cleaned out at reboots.
> >
> > Also, other things to consider:
> >
> > 1. OVS stores certs under /var/lib/openvswitch and /etc/openvswitch
> > directories. Would storing these under those be good?
> > 2. When Controller certs are provided to Faucet, they must have the same
> > permission of user who is running faucet service - normally, this could be
> > "root" or "faucet". If faucet, then chowm cert files to facuet:faucet (if
> > faucet is user and faucet is group)
> > 3. One other thing to add is if we have more than 1 switch connecting
> > via TLS to the same controller, how to add certs to the 2nd switch so that
> > it can work with the already running controller.
> >
> >
> > Thanks
> >
> > On Thu, Jul 13, 2017 at 6:04 PM, Josh Bailey <notifications@github.com>
> > wrote:
> >
> > > Merged #756 <https://github.com/REANNZ/faucet/pull/756>.
> > >
> > > —
> > > You are receiving this because you are subscribed to this thread.
> > > Reply to this email directly, view it on GitHub
> > > <https://github.com/REANNZ/faucet/pull/756#event-1163570955>, or mute the
> > > thread
> > > <https://github.com/notifications/unsubscribe-auth/ABmf82YJ9A9lTWECwxeoR6Bf8LoafEL1ks5sNr6CgaJpZM4OXr4X>
> > > .
> > >
> >
> > —
> > You are receiving this because you authored the thread.
> > Reply to this email directly, view it on GitHub, or mute the thread.
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub, or mute the thread.
>
